lib/client.ts

```diff
@@ -115,8 +115,19 @@ export class Client {
       () => notNil(params.id),
       collectionUrl => `${collectionUrl}/${params.id}`
     ),
-    (hostname) => `${hostname}${S(base).template(params, '{', '}').s}`
+    (hostname) => `${hostname}/${this.version(api_version)}${S(base).template(params, '{', '}').s}`
   )(this.host)
+
+  private version = (api_version: string): string => {
+    api_version = (api_version === 'undefined' || api_version === '') ? 'v1.0' : api_version;
+    if (api_version === 'vapid' || api_version === 'vapid/') {
+      return api_version
+    } else if(api_version.match(/v\d+\.\d+/)) {
+      return `rest/${api_version}`
+    } else {
+      throw new Error(`'${api_version}' is an invalid Procore API version`)
```

Comment:
As discussed, it would be nice to catch it o the `js-sdk-sample-app` side and show some alert/popup/banner message.